modules/rds/main.tf

```diff
@@ -1,84 +1,172 @@
-resource "aws_db_subnet_group" "postgres_subnet_group" {
-  name       = "${var.prefix}-postgres-subnet-group"
-  subnet_ids = var.subnet_ids
+## Terraform Requirements
+terraform {
+  required_providers {
+    aws = {
+      source  = "hashicorp/aws"
+      version = ">= 3.5.0"
+    }
+  }
+}
+
+
+## AWS Provider Settings
+provider "aws" {
+  region  = var.region
+  profile = var.aws_profile
 }
 
-# Ignore aws profile in case this was deployed with CI or
-# in a machine without aws profile defined
+
+## Local Variables
 locals {
-  used_profile = contains([var.profile], "default") ? "" : "--profile ${var.profile}"
+  tags = merge(var.tags, { Deployment = var.prefix })
+  # Ignore aws profile in case this was deployed with CI or in a machine without
+  # aws profile defined
+  used_profile = contains([var.aws_profile], "default") ? "" : "--profile ${var.aws_profile}"
 }
+
+
+## Resources
+
+## =============================================================================
+## DATABASE RESOURCES
+## =============================================================================
+
+## postgres_subnet_group - Subnet Group the PostgreSQL database is associated with.
+## =============================================================================
+resource "aws_db_subnet_group" "postgres_subnet_group" {
+  ## REQUIRED
+  subnet_ids = var.lambda_subnet_ids
+
+  ## OPTIONAL
+  description = "ORCA PostgreSQL Subnet group where the ORCA database resides."
+  name        = "${var.prefix}-postgres-subnet-group"
+  tags        = local.tags
+}
+
+
+## postgresql - PostgreSQL database created for ORCA metadata/status
+## =============================================================================
 # https://blog.faraday.io/how-to-create-an-rds-instance-with-terraform/
+# https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/db_instance
+# https://registry.terraform.io/modules/terraform-aws-modules/rds/aws/latest
 resource "aws_db_instance" "postgresql" {
+  ## REQUIRED
+  allocated_storage = 20
+  engine            = "postgres"
+  identifier        = "${var.prefix}-postgres-sndbx"
+  instance_class    = "db.t3.micro"
+  password          = var.postgres_user_pw
+  username          = "postgres"
```

Comment:
should we move some of these parameters to variables.tf and then use var.xx  here? Just a thought.

Reply:
So the short answer is probably not as changing anything but the instance_class can have some negative impacts to our current code base. For the most part I kept the resource the same and just documented items. The one change I did make was moving from a db.t2.micro to a db.t3.micro. The reason I did this was the cost was the same and I could enable encryption which is not available with the db.t2.micro.

Realistically ORCA in it's current form is pretty light on the database and anything bigger than a micro is wasted. In addition, in the next PI (only a few weeks away) we will be working on moving over the the Cumulus RDS instance so this resource will likely be removed completely from our terraform module since we will rely on their instance. Also, LP and PO DAAC have been using an incarnation of this database for over a year with no issues and no requirements to expose those items.

That being said, if we were going to maintain our own database instance long term, I would surface items like instance_class. If you feel strongly about surfacing that variable for the short term let me know.